### ISSUE TITLE
WiFi static buffer num error (IDFGH-7667)

### ISSUE THREAD

**Issue by kaspernyhus** [espressif/esp-idf#9217]
## Environment

- Development Kit:     ESP32S3-DevKitC

- Module or chip used:  ESP32-WROOM-32
- IDF version ESP-IDF v4.4.1-dirty
- Build System:         idf.py
- Compiler version xtensa-esp32-elf-gcc (crosstool-NG esp-2021r2-patch3) 8.4.0
- Operating System:     Linux
- Using an IDE?: VScode
- Power Supply:         USB
- 
## Problem Description
In menuconfig the maximum number of static WiFi TX buffers should be 64, but if I'm allocating more than 32 (e.g. 33), the esp crashes when booting

// more context
https://www.esp32.com/viewtopic.php?f=13&t=28284&sid=6787639b8f477954310179897e3ad9e7
https://www.esp32.com/viewtopic.php?f=13&t=27665

### Expected Behavior
Allocation of more static WiFi TX buffes
### Actual Behavior
´W (450) wifi:esf_buf_setup_static: alloc eb fail(1)´
Crash with
ESP_ERR_NO_MEM 
### Steps to reproduce
1. Component config --> Wi-Fi --> Type of WiFi TX buffers --> Static
2. --> Max number of WiFi static TX buffers --> 33
3. build and run

## Debug Logs
```
I (29) boot: ESP-IDF v4.4.1-dirty 2nd stage bootloader
I (29) boot: compile time 14:25:48
I (29) boot: chip revision: 0
I (31) boot.esp32s3: Boot SPI Speed : 80MHz
I (36) boot.esp32s3: SPI Mode       : DIO
I (41) boot.esp32s3: SPI Flash Size : 2MB
I (46) boot: Enabling RNG early entropy source...
I (51) boot: Partition Table:
I (55) boot: ## Label            Usage          Type ST Offset   Length
I (62) boot:  0 nvs              WiFi data        01 02 00009000 00006000
I (69) boot:  1 phy_init         RF data          01 01 0000f000 00001000
I (77) boot:  2 factory          factory app      00 00 00010000 00100000
I (84) boot: End of partition table
I (88) esp_image: segment 0: paddr=00010020 vaddr=3c080020 size=169dch ( 92636) map
I (113) esp_image: segment 1: paddr=00026a04 vaddr=3fc94800 size=04164h ( 16740) load
I (117) esp_image: segment 2: paddr=0002ab70 vaddr=40374000 size=054a8h ( 21672) load
I (124) esp_image: segment 3: paddr=00030020 vaddr=42000020 size=73308h (471816) map
I (212) esp_image: segment 4: paddr=000a3330 vaddr=403794a8 size=0b358h ( 45912) load
I (223) esp_image: segment 5: paddr=000ae690 vaddr=50000000 size=00010h (    16) load
I (230) boot: Loaded app from partition at offset 0x10000
I (230) boot: Disabling RNG early entropy source...
I (243) cpu_start: Pro cpu up.
I (243) cpu_start: Starting app cpu, entry point is 0x403751d8
0x403751d8: call_start_cpu1 at /home/kaspernyhus/esp/esp-idf/components/esp_system/port/cpu_start.c:160

I (221) cpu_start: App cpu up.
I (257) cpu_start: Pro cpu start user code
I (257) cpu_start: cpu freq: 240000000
I (257) cpu_start: Application information:
I (260) cpu_start: Project name:     restful_api_test
I (266) cpu_start: App version:      a9cf556-dirty
I (271) cpu_start: Compile time:     Jun 23 2022 11:03:09
I (277) cpu_start: ELF file SHA256:  fcebd4ece5f4d854...
I (283) cpu_start: ESP-IDF:          v4.4.1-dirty
I (289) heap_init: Initializing. RAM available for dynamic allocation:
I (296) heap_init: At 3FC9C400 len 00043C00 (271 KiB): D/IRAM
I (302) heap_init: At 3FCE0000 len 0000EE34 (59 KiB): STACK/DRAM
I (309) heap_init: At 3FCF0000 len 00008000 (32 KiB): DRAM
I (315) heap_init: At 600FE000 len 00002000 (8 KiB): RTCRAM
I (322) spi_flash: detected chip: gd
I (326) spi_flash: flash io: dio
W (330) spi_flash: Detected size(8192k) larger than the size in the binary image header(2048k). Using the size in the binary image header.
I (343) sleep: Configure to isolate all GPIO pins in sleep state
I (350) sleep: Enable automatic switching of GPIO sleep configuration
I (357) cpu_start: Starting scheduler on PRO CPU.
I (0) cpu_start: Starting scheduler on APP CPU.
I (400) pp: pp rom version: e7ae62f
I (400) net80211: net80211 rom version: e7ae62f
I (410) wifi:wifi driver task: 3fce5a40, prio:23, stack:6656, core=0
I (410) system_api: Base MAC address is not set
I (410) system_api: read default base MAC address from EFUSE
I (420) wifi:wifi firmware version: 63017e0
I (420) wifi:wifi certification version: v7.0
I (420) wifi:config NVS flash: enabled
I (430) wifi:config nano formating: disabled
I (430) wifi:Init data frame dynamic rx buffer num: 32
I (440) wifi:Init management frame dynamic rx buffer num: 32
I (440) wifi:Init management short buffer num: 32
W (450) wifi:esf_buf_setup_static: alloc eb fail(1)
I (450) wifi:Deinit lldesc rx mblock:0
I (450) wifi:Deinit lldesc rx mblock:0
I (460) wifi:Deinit lldesc rx mblock:0
I (470) wifi_init: rx ba win: 6
I (470) wifi_init: tcpip mbox: 32
I (470) wifi_init: udp mbox: 6
I (470) wifi_init: tcp mbox: 6
I (480) wifi_init: tcp tx win: 5744
I (480) wifi_init: tcp rx win: 5744
I (480) wifi_init: tcp mss: 1440
I (490) wifi_init: WiFi IRAM OP enabled
I (490) wifi_init: WiFi RX IRAM OP enabled
ESP_ERROR_CHECK failed: esp_err_t 0x101 (ESP_ERR_NO_MEM) at 0x4037bc0c
0x4037bc0c: _esp_error_check_failed at /home/kaspernyhus/esp/esp-idf/components/esp_system/esp_err.c:42

file: "../main/wifi_config.c" line 51
func: wifi_init_sta
expression: esp_wifi_init(&cfg)

abort() was called at PC 0x4037bc0f on core 0
0x4037bc0f: _esp_error_check_failed at /home/kaspernyhus/esp/esp-idf/components/esp_system/esp_err.c:43



Backtrace:0x40375aa6:0x3fcf3c400x4037bc19:0x3fcf3c60 0x40382522:0x3fcf3c80 0x4037bc0f:0x3fcf3cf0 0x42007425:0x3fcf3d10 0x4200722b:0x3fcf3ea0 0x42073001:0x3fcf3ec0 0x4037ee29:0x3fcf3ee0 
0x40375aa6: panic_abort at /home/kaspernyhus/esp/esp-idf/components/esp_system/panic.c:402

0x4037bc19: esp_system_abort at /home/kaspernyhus/esp/esp-idf/components/esp_system/esp_system.c:128

0x40382522: abort at /home/kaspernyhus/esp/esp-idf/components/newlib/abort.c:46

0x4037bc0f: _esp_error_check_failed at /home/kaspernyhus/esp/esp-idf/components/esp_system/esp_err.c:43

0x42007425: wifi_init_sta at /home/kaspernyhus/esp/esp32_sandbox/rest_api_test/build/../main/wifi_config.c:51 (discriminator 1)

0x4200722b: app_main at /home/kaspernyhus/esp/esp32_sandbox/rest_api_test/build/../main/main.c:85 (discriminator 2)

0x42073001: main_task at /home/kaspernyhus/esp/esp-idf/components/freertos/port/port_common.c:129 (discriminator 2)

0x4037ee29: vPortTaskWrapper at /home/kaspernyhus/esp/esp-idf/components/freertos/port/xtensa/port.c:131
```

**Comment by kaspernyhus** [comment]
any thoughts on this issue?

**Comment by MaxwellAlan** [comment]
Hi @kaspernyhus ,
There is a limit in wifi lib, we will fix this.
Thanks.

**Comment by Alvin1Zhang** [comment]
Thanks for reporting, fix on master branch is available at https://github.com/espressif/esp-idf/commit/77a7fa77e2926963189b37862e789a85686f8e28, we will backport the fix to release/4.4, thanks.
